### PR TITLE
Respect CSS zoom

### PR DESCRIPTION
**Pull request by ebads67**
Css `zoom` property affects the mouse position in `event`s. However, functions like `getBoundingClientRect` do not care about the css `zoom`. Hence, the calculations that use mouse positions and `getBoundingClientRect` return wrong values. In this PR those calculations are fixed to return the correct value when css `zoom` property is used.

Css `zoom` has only been implemented in Nightly version of Firefox. So the fix does not apply to Firefox.

In all the examples below the css `zoom` property is set to 2 on the body of the document..

Issue with checklist interactions:
checklist-zoom
Issue with drag handle:
drag-handle-zoom
Issue with context menu:
context-menu-zoom

### CODE REVIEW

**Comment by ivailop7** [comment]
Looking very good! Thanks for fixing! Integrity fails because of prettier formatting.

**Review by ivailop7 — APPROVED** [review]
Fantastic fix!

**Review by zurfyx — APPROVED** [review]

File: packages/lexical-utils/src/index.ts [hunk 1]
```diff
@@ -524,3 +525,20 @@ export function $insertFirst(parent: ElementNode, node: LexicalNode): void {
...
   }
 }
+
+/**
+ * Calculates the zoom level of an element as a result of using
+ * css zoom property.
+ * @param element
+ */
+export function calculateZoomLevel(element: Element | null): number {
+  if (IS_FIREFOX) {
```
Comment (Daniel15) [inline packages/lexical-utils/src/index.ts:535]: Should this do feature detection rather than browser detection?

**Comment by Daniel15** [comment]
> Css zoom has only been implemented in Nightly version of Firefox.

Interesting, I didn't realise they finally implemented it. The bug tracking its implementation is 17 years old: https://bugzilla.mozilla.org/show_bug.cgi?id=390936

**Pull request merged**